### PR TITLE
Added SIDECAR_HTTP_PROXY context on Envoy Filter

### PR DESCRIPTION
**Pull request by alandiegosantos**
Creating the context for the HTTP_PROXY port on EnvoyFilter, allowing users to define custom configurations for the HTTP_PROXY port. 

Relates to istio/istio#38673

### CODE REVIEW

File: networking/v1alpha3/envoy_filter.proto [hunk 1]
```diff
@@ -488,6 +488,10 @@ message EnvoyFilter {
 
     // Gateway listener/route/cluster.
     GATEWAY = 3;
+
+    // HTTP Proxy listener/route.
+    SIDECAR_HTTP_PROXY = 4;
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply networking/v1alpha3/envoy_filter.proto:493]: I get we need to change Istio implementation to support this, just not sure if API needs changed as well

[the remark this answers is not in the document]
Comment (alandiegosantos) [reply networking/v1alpha3/envoy_filter.proto:493]: The way I see, the HTTP Proxy listener is a special listener on Istio. To add specific patches for routes request requires knowledge about the route name (http_proxy at the moment). So, changing the name in the future might cause compatibility issues. But, given patches are low-level configuration, the compatibility between minor versions is might be not required. 
I do not have a strong opinion about it, so I think we can start using the SIDECAR_OUTBOUND context for the moment and revisit this decision in the future if necessary.

**Comment by costinm** [comment]
There are 3 special listeners - sidecar inbound and outbound plus http
proxy. All 3 capture multiple virtual hosts, with associated settings and
routing.
In addition we have 'regular' listeners, with regular IP:ports.

We have a context for 2 of the special listeners - and we have one for
'regular' - and we are missing one for the http_proxy.

I think the problem with using SIDECAR_OUTBOUND is that you can't make a
change only on one of the 2 special listeners: routes, virtual
hosts are duplicated. But if it works for your use case - it may be better
to drop this PR.

Can you describe your use case - what are you trying to patch for the http
proxy ?

I think EnvoyFilter is in dire need of replacement with higher level APIs
and 'inverting' - instead of patching the config with arbitrary json we
should identify stable envoy filters we want to support, define validated
CRDs for them and use policy attachment. .

**Comment by alandiegosantos** [comment]
@costinm 
The use-case is to enable WASM filters in the HTTP proxy port. It can be done with the SIDECAR_OUTBOUND context. So, we can close this MR and re-open it in the future.

**Pull request closed without merging**